drivers/sensor/bmi270/bmi270.c

```diff
@@ -0,0 +1,714 @@
+/*
+ * Copyright (c) 2020 Bosch Sensortec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT bosch_bmi270
+
+#include <drivers/i2c.h>
+#include <init.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+
+#include "bmi270.h"
+#include "bmi270_config_file.h"
+
+LOG_MODULE_REGISTER(bmi270, CONFIG_SENSOR_LOG_LEVEL);
+
+#define MSB_TO_DATA(lsb, msb) (int16_t)(((uint16_t) msb << 8) + (uint16_t)lsb)
+
+#ifndef BMI270_WR_LEN
+#define BMI270_WR_LEN 256
+#endif
+
+static int reg_read(uint8_t reg, uint8_t *data, uint16_t length,
+		    struct bmi270_data *dev)
+{
+	int ret = 0;
+
+	if ((dev->i2c != NULL) && (data != NULL)) {
+		ret = i2c_burst_read(dev->i2c, dev->i2c_addr, reg, data,
+				     length);
+	} else {
+		ret = -EINVAL;
+	}
+
+	return ret;
+}
+
+static int reg_write(uint8_t reg, const uint8_t *data, uint16_t length,
+		     struct bmi270_data *dev)
+{
+	int ret = 0;
+
+	if ((dev->i2c != NULL) && (data != NULL)) {
+		ret = i2c_burst_write(dev->i2c, dev->i2c_addr, reg, data,
+				      length);
+	} else {
+		ret = -EINVAL;
+	}
+
+	return ret;
+}
+
+static void channel_accel_convert(struct sensor_value *val, int64_t raw_val,
+				  uint8_t range)
+{
+	/* 16 bit accelerometer. 2^15 bits represent the range in G */
+	/* Converting from G to m/s^2 */
+	raw_val = (raw_val * SENSOR_G * (int64_t) range) / 32768LL;
+
+	val->val1 = raw_val / 1000000;
+	val->val2 = raw_val % 1000000;
+
+	/* Normalize val to make sure val->val2 is positive */
+	if (val->val2 < 0) {
+		val->val1 -= 1;
+		val->val2 += 1000000;
+	}
+}
+
+static void channel_gyro_convert(struct sensor_value *val, int64_t raw_val,
+				 uint16_t range)
+{
+	/* 16 bit gyroscope. 2^15 bits represent the range in degrees/s */
+	/* Converting from degrees/s to radians/s */
+
+	val->val1 = ((raw_val * (int64_t) range * SENSOR_PI)
+		     / (180LL * 32768LL)) / 1000000LL;
+	val->val2 = ((raw_val * (int64_t) range * SENSOR_PI)
+		     / (180LL * 32768LL)) % 1000000LL;
```

Comment:
consider using already provided `sensor_degrees_to_rad` function in the sensor API